pkg/cluster/loadbalancerprofile.go

```diff
@@ -0,0 +1,386 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-08-01/network"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+	"github.com/Azure/ARO-RP/pkg/util/uuid"
+)
+
+func (m *manager) reconcileLoadBalancerProfile(ctx context.Context) error {
+	if m.doc.OpenShiftCluster.Properties.NetworkProfile.OutboundType != api.OutboundTypeLoadbalancer {
+		return nil
+	}
+
+	resourceGroupName := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+	infraID := m.doc.OpenShiftCluster.Properties.InfraID
+
+	lb, err := m.loadBalancers.Get(ctx, resourceGroupName, infraID, "")
+	if err != nil {
+		return err
+	}
+
+	err = m.reconcileOBRuleV4OBIPs(ctx, lb)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reconcile the outbound rule "outbound-rule-v4" frontend IP Config.
+func (m *manager) reconcileOBRuleV4OBIPs(ctx context.Context, lb mgmtnetwork.LoadBalancer) error {
+	m.log.Info("reconciling outbound-rule-v4")
+	defer func() {
+		err := m.deleteUnusedManagedIPs(ctx)
+		if err != nil {
+			m.log.Error("failed to cleanup unused managed IPs, error: %w", err)
+		}
+	}()
```

Comment:
this might update IPs, but not patch cosmos EffectiveIPs, right? What happens when we hit the first error case, and we can't run patchEffectiveOutboundIPs for the first time, then we attempt to delete managedIPs?

Reply:
What is "this" in your question?  If you are talking about the process as a whole, the load balancer could be updated with a change in managed IPs but then cosmos patch may fail.  This would run and cleanup unused managed IPs but the value of `effectiveOutboundIPs` would be wrong.  If "this" is the `m.deleteUnusedManagedIPs` then only IPs that are not attached to the loadbalancer are cleaned up.  

Overall this tries to prevent orphaning IP addresses if the process fails at any point.

Reply:
sorry - "this" meaning the entire `defer`. The path I outline is:

1. New cluster, need to patch cosmos as step 1
2. Cosmos write fails (e.g., cosmos is unavailabe?), returns an error, but before we can return, we do this `defer` statement

Does that path make sense to execute on?